CI: Builds should terminate if configure.sh fails instead of moving to build phase

Right now if `./tools/configure.sh` fails the build will continue on to the make phase which will cause all kind of unhelpful error messages.

For example this recent case was ignored because the change looked unrelated to the PR based on the compiler errors:

```
 + /github/workspace/sources/nuttx/tools/testbuild.sh -j 2 -e '-Wno-cpp -Werror' testlist/arm-11.dat
====================================================================================
Configuration/Tool: stm32f4discovery/hciuart,CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
boards/arm/sama5/giant-board/Kconfig:110: syntax error
boards/arm/sama5/giant-board/Kconfig:109: invalid option
make: *** [tools/Makefile.unix:471: olddefconfig] Error 1
  Enabling CONFIG_ARMV7M_TOOLCHAIN_GNU_EABIL
boards/arm/sama5/giant-board/Kconfig:110: syntax error
boards/arm/sama5/giant-board/Kconfig:109: invalid option
make: *** [tools/Makefile.unix:471: olddefconfig] Error 1
  Building NuttX...
In file included from /github/workspace/sources/nuttx/include/arch/chip/irq.h:49,
                 from /github/workspace/sources/nuttx/include/arch/irq.h:38,
                 from /github/workspace/sources/nuttx/include/nuttx/irq.h:53,
                 from /github/workspace/sources/nuttx/include/nuttx/sched.h:40,
                 from /github/workspace/sources/nuttx/include/sched.h:34,
                 from /github/workspace/sources/nuttx/include/spawn.h:47,
                 from ./exec_builtin.c:49:
##[error]/github/workspace/sources/nuttx/include/arch/stm32/chip.h:96:4: error: #error "Only one STM32 family must be selected !"
```

--- a/tools/configure.sh
+++ b/tools/configure.sh
@@ -33,6 +33,8 @@
 # POSSIBILITY OF SUCH DAMAGE.
 #
 
+set -e
+
 WD=`test -d ${0%/*} && cd ${0%/*}; pwd`
 TOPDIR="${WD}/.."
 USAGE="

--- a/tools/sethost.sh
+++ b/tools/sethost.sh
@@ -17,6 +17,8 @@
 # under the License.
 #
 
+set -e
+
 progname=$0
 host=
 wenv=

